scenes/game/local_game.gd

```diff
@@ -1396,6 +1400,69 @@ class Player:
 			buddy_index = buddy_id_to_index[buddy_id]
 		buddy_locations[buddy_index] = new_location
 
+	func get_next_free_buddy_id():
+		for i in range(buddy_locations.size()):
+			if buddy_locations[i] == -1:
+				return buddy_id_to_index.keys()[i]
+		return ""
+
+	func get_buddy_id_at_location(location : int):
+		for i in range(buddy_locations.size()):
+			if buddy_locations[i] == location:
+				return buddy_id_to_index.keys()[i]
+		return ""
+
+	func get_buddies_in_play():
+		var buddies = []
+		for i in range(buddy_locations.size()):
+			if buddy_locations[i] != -1:
+				buddies.append(buddy_id_to_index.keys()[i])
+		return buddies
+
+	func get_buddies_on_opponent():
+		var opposing_player = parent._get_player(parent.get_other_player(my_id))
+		var matching_buddies = []
+		for i in range(buddy_locations.size()):
+			if opposing_player.is_in_location(buddy_locations[i]):
+				matching_buddies.append(buddy_id_to_index.keys()[i])
+		return matching_buddies
+
+	func get_buddies_adjacent_opponent():
+		var opposing_player = parent._get_player(parent.get_other_player(my_id))
+		var matching_buddies = []
+		for i in range(buddy_locations.size()):
+			var location = buddy_locations[i]
+			if not opposing_player.is_in_location(location):
+				if opposing_player.is_in_location(location - 1) or opposing_player.is_in_location(location + 1):
+					matching_buddies.append(buddy_id_to_index.keys()[i])
+		return matching_buddies
+
+	func count_buddies_between_opponent():
+		var opposing_player = parent._get_player(parent.get_other_player(my_id))
+		var count = 0
+		var started = false
+		for location in range(MinArenaLocation, MaxArenaLocation + 1):
```

Comment:
I think I defined "is_left_of_location" and similar functions, if that makes this logic any easier